.github/workflows/pre-release.yml

```diff
@@ -27,8 +21,8 @@ jobs:
     steps:
     - name: Check out the code
       uses: actions/checkout@v3
-      with:
-        ref: ${{ github.event.inputs.commit }}
+    - name: Switch to commit ${{ github.event.inputs.commit }}
+      run: git checkout ${{ github.event.inputs.commit }}
```

Comment:
Manual commit switch seems to be more reliable, based on my testing.

Reply:
More reliable in what sense?
Did you notice failures when using actions/checkout?

Reply:
I tried using `ref` again, it consistently failed on me (not sure what I'm missing here):

![image](https://user-images.githubusercontent.com/7288/202768208-7dc4bebf-a4fa-4f7a-a017-03af6ebf42d2.png)

Reply:
Oh, it **might be** because you're not supplying the full SHA?
In all of my testing, I was always copying the full SHA and used it to trigger the workflow.

This is of secondary importance.
I'm happy to use either.

Reply:
IF that's the reason why, maybe it's not even bad to use that mechanism as an enforcing function because the SHA itself is going to be used in an image name.

That way, we can avoid having multiple images for the same commit. E.g.
- `metabase/metabase: 3e518549f803ff050fe6ae43ddb6a003908b18a0`
- `metabase/metabase: 3e5185`
- `metabase/metabase: 3e518549f803ff050fe6ae43ddb6a0039`

Reply:
^ Either that or we need to have an additional logic that validates the SHA and possibly truncates it to a minimum required length, regardless of the input provided?

Reply:
That did the trick, thank you!
It's very strange since the canonical `git` can work with any length of SHA1 (as long as it's unique).

Reply:
I like the idea of limiting the length of the SHA1 hash to just 8 characters (more than sufficient, for a while).

Reply:
^ https://github.com/metabase/metabase/pull/26597/commits/0e73ec3699f7e32c8e32338d6d6c78106accc3a4 should do the trick.

And I switched back to implicitly enforcing full SHA-1 as an input in https://github.com/metabase/metabase/pull/26597/commits/3039a30d544564ce0720f8a709549de9858c67d4.

Feel free to change/adapt any part of it as you see fit.